pathrs-lite/internal/linux/openat2_linux.go

```diff
@@ -13,12 +13,12 @@ package linux
 
 import (
 	"golang.org/x/sys/unix"
-
-	"github.com/cyphar/filepath-securejoin/pathrs-lite/internal/gocompat"
 )
 
 // HasOpenat2 returns whether openat2(2) is supported on the running kernel.
-var HasOpenat2 = gocompat.SyncOnceValue(func() bool {
+// This value is *not* cached because it may change if the program applies a
+// new seccomp-bpf filter.
+var HasOpenat2 = func() bool {
```

Comment:
Since we're not using `sync.OnceValue`, we can drop the `var` stuff, too. IOW
```suggestion
func HasOpenat2() bool {
```

As a nice side effect, this will improve package documentation (godoc does not render variable documentation nicely).